src/expression.rs

```diff
@@ -573,6 +571,7 @@ impl<F: Field> BinaryExpression<F> {
 
     /// Join these bits into the field element they encode. This method requires that
     /// `2^self.len() < |F|`, otherwise the result might not fit in a single field element.
+    //TODO: Is it possible for the field F to be of order 2^n + 1 (Fermat prime) for some n, and if so can we then conclude that all elements can be fully encoded by a n-binary string? If so then self.len() < Element::<F>::max_bits() needs to be checked for LEQ for this specific case. Note that fields of order order 2^n + 1 make the corresponding modular operations very inefficient so this is likely not going to be the case.
```

Comment:
I plan to ask this question to ZKP experts as it could, in theory, make operations much easier for such fields.